sonar-plugin-api/src/main/java/org/sonar/api/batch/sensor/internal/SensorContextTester.java

```diff
@@ -210,30 +211,33 @@ public NewIssue newIssue() {
   }
 
   @CheckForNull
-  public Integer lineHits(String fileKey, CoverageType type, int line) {
-    DefaultCoverage defaultCoverage = sensorStorage.coverageByComponentAndType.get(fileKey, type);
-    if (defaultCoverage == null) {
-      return null;
-    }
-    return defaultCoverage.hitsByLine().get(line);
+  public Integer lineHits(String fileKey, int line) {
+    Optional<Integer> sum = sensorStorage.coverageByComponent.get(fileKey).stream()
+      .map(c -> c.hitsByLine().get(line))
+      .flatMap(i -> Stream.of(Optional.ofNullable(i)))
+      .reduce(Optional.empty(),
+        (o1, o2) -> o1.isPresent() ? o1.map(i -> i + o2.orElse(0)) : o2);
+    return sum.orElse(null);
```

Comment:
How about not using `Optional`:

    return ...
      .flatMap(v -> Stream.of(v))
      .filter(v -> v != null)
      .reduce(null, (o1, o2) -> o1 == null ? o2 : o1 + o2);

Also, there's much duplication between the methods, a helper function could be easier to maintain.